src/search/parser/lexical_analyzer.cc

```diff
@@ -0,0 +1,131 @@
+#include "lexical_analyzer.h"
+
+#include "errors.h"
+
+#include "../utils/collections.h"
+#include "../utils/system.h"
+
+#include <cassert>
+#include <regex>
+
+using namespace std;
+
+namespace parser {
+static regex build_regex(const string &s, regex::flag_type f = regex_constants::ECMAScript) {
+    return regex("^\\s*(" + s + ")\\s*", f);
+}
+
+static const vector<pair<TokenType, regex>> token_type_expressions = {
+    {TokenType::OPENING_PARENTHESIS, build_regex("\\(")},
+    {TokenType::CLOSING_PARENTHESIS, build_regex("\\)")},
+    {TokenType::OPENING_BRACKET, build_regex("\\[")},
+    {TokenType::CLOSING_BRACKET, build_regex("\\]")},
+    {TokenType::COMMA, build_regex(",")},
+    {TokenType::EQUALS, build_regex("=")},
+    {TokenType::FLOAT, build_regex(
+       R"([+-]?((\d*\.\d+|\d+\.)(e[+-]?\d+|[kmg])?)\b)",
+       regex_constants::ECMAScript | regex_constants::icase)},
+    {TokenType::INTEGER, build_regex(
+       R"([+-]?(infinity|\d+(e+?\d+|[kmg])?)\b)",
```

Comment:
Currently, this is not a problem because we can parse an integer token as a float (without first parsing it as an `int`). But if we want to detect the type of an expression from the expression rather than the context, this might change.

I'll replace the expression by `[+-]?(infinity|\d+([kmg]\b)?)`.

Reply:
I'll resolve this conversation with the change above. The parsing also need to be updated but that was not consistent anyway, and there is a TODO about it at that place, so we won't miss it.